crates/cli-support/src/js/mod.rs

```diff
@@ -2634,63 +2635,115 @@ impl<'a, 'b> SubContext<'a, 'b> {
+        let method = |c: &mut ExportedClass, docs, fn_name, fn_prfx, js, ts| {
+            c.contents.push_str(docs);
+            c.contents.push_str(fn_prfx);
+            c.contents.push_str(fn_name);
+            c.contents.push_str(js);
+            c.contents.push_str("\n");
+            c.typescript.push_str(docs);
+            c.typescript.push_str("  "); // Indentation
+            c.typescript.push_str(fn_prfx);
+            c.typescript.push_str(ts);
+            c.typescript.push_str("\n");
+        };
+        let setter_from_method_call = |c: &mut ExportedClass, docs, fn_name, js: &str, s_name| {
+            let fn_args = {
+                let start = js.find('(').unwrap();
+                let end = js[start..].find(')').unwrap();
+                &js[start..=end]
+            };
+            c.contents.push_str(docs);
+            c.contents.push_str("set ");
+            c.contents.push_str(s_name);
+            c.contents.push_str(fn_args);
+            c.contents.push_str(" {\n    this.");
+            c.contents.push_str(fn_name);
+            c.contents.push_str(fn_args);
+            c.contents.push_str(";\n}\n");
+        };
+        let finish_j2r = |mut j2r: Js2Rust| -> Result<(_, _, _), Error> {
+            Ok(j2r
+                .process(descriptor.unwrap_function(), &export.function.arg_names)?
+                .finish(
+                    "",
+                    &format!("wasm.{}", wasm_name),
+                    ExportedShim::Named(&wasm_name),
+                ))
+        };
+        match &export.method_kind {
+            decode::MethodKind::Constructor => {
+                fn_name = "constructor";
+                let mut j2r = Js2Rust::new(fn_name, self.cx);
+                j2r.constructor(Some(class_name));
+                let (js, ts, raw_docs) = finish_j2r(j2r)?;
+                let class = require_class(&mut self.cx.exported_classes, class_name);
+                if class.has_constructor {
+                    bail!("found duplicate constructor `{}`", export.function.name);
+                }
+                class.has_constructor = true;
+                let docs = docs(raw_docs);
+                method(class, &docs, fn_name, "", &js, &ts);
+                Ok(())
+            }
+            decode::MethodKind::Operation(operation) => {
+                let mut j2r = Js2Rust::new(fn_name, self.cx);
+                let mut static_prfx = "";
+                if operation.is_static {
+                    static_prfx = "static ";
+                } else {
+                    j2r.method(export.consumed);
+                }
+                let (js, ts, raw_docs) = finish_j2r(j2r)?;
+                let class = require_class(&mut self.cx.exported_classes, class_name);
+                let docs = docs(raw_docs);
+                match operation.kind {
+                    decode::OperationKind::Getter(maybe_getter_name) => {
+                        if export.function.renamed_via_js_name {
```

Comment:
Could a comment be added here explaining what's going on?

Reply:
I'm also not sure if this is the best way to approach `js_name`? I would think that we basically just ignore `js_name` on setters/getters because customizing the field name would be `#[wasm_bindgen(getter = otherName)]` or something like that. If a new method wants to be created I think we'd want to do that through a dedicated separate method in the source

Reply:
> If a new method wants to be created I think we'd want to do that through a dedicated separate method in the source

This is going to be wonderful and life is also going to be very much easier but I think that we shouldn't ignore the `js_name` attribute in getters/setters, that would lead to unexpected behavior, confusing users. Instead, we could forbid both declarations in the same function or simply remove the ability to define names in getters/setters (the name would de defined in `js_name`).

Reply:
I'd also be fine considering `getter = foo` equivalent to `getter, js_name = foo` and maybe producing an error for something like `getter = foo, js_name = bar`, what do you think?

FWIW in the long run I'd like to turn "unused attribute warnings" into hard errors by default, but while we have a crate feature for that it'd be a breaking change so I don't think we can do it just yet.

Reply:
> it'd be a breaking change so I don't think we can do it just yet.

So unfortunate :(
Ok, the best approach right now would be to ignore the `js_name` attribute.
**Table updated**.